### PR TITLE
feat(backend): logging events

### PR DESCRIPTION
**Pull request by ulfgebhardt**
## 🍰 Pullrequest

logging events

### Issues

- contributes to #143 

### Todo

- [x] Test fixes
- [x] Test for events to be written to database

### CODE REVIEW

File: backend/src/graphql/inputs/ContactFormInput.ts [hunk 1]
```diff
@@ -4,7 +4,7 @@ import { InputType, Field } from 'type-graphql'
 // values according to prisma/schema.prisma
 const MAX_CHARS_FIRST_NAME = 50
 const MAX_CHARS_LAST_NAME = 50
-const MAX_CHARS_EMAIL = 255
+const MAX_CHARS_EMAIL = 254
```
Comment (Elweyn) [inline backend/src/graphql/inputs/ContactFormInput.ts:7]: Duplicate with inputs/SubscribeToNewsletterInput.ts consts.
Could be interessting to have it in one file for the case where we need to update it to other values

File: backend/src/graphql/inputs/SubscribeToNewsletterInput.ts [hunk 2]
```diff
@@ -4,7 +4,7 @@ import { InputType, Field } from 'type-graphql'
 // values according to prisma/schema.prisma
 const MAX_CHARS_FIRST_NAME = 50
 const MAX_CHARS_LAST_NAME = 50
-const MAX_CHARS_EMAIL = 255
+const MAX_CHARS_EMAIL = 254
```
Comment (Elweyn) [inline backend/src/graphql/inputs/SubscribeToNewsletterInput.ts:7]: Duplicate with inputs/SubscribeToNewsletterInput.ts consts.
Could be interessting to have it in one file for the case where we need to update it to other values

File: backend/src/graphql/resolvers/ContactFormResolver.ts [hunk 3]
```diff
@@ -2,19 +2,18 @@ import { Resolver, Mutation, Query, Arg } from 'type-graphql'
...
+  createContactForm(@Arg('contactFormData') contactFormData: ContactFormInput): boolean {
+    void sendContactEmails(contactFormData)
+    void EVENT_CONTACTFORM_SEND(contactFormData.email)
     return true
   }
 
+  // TODO: remove - see https://github.com/MichalLytek/type-graphql/issues/301#issuecomment-480046611
   // needed to avoid: GraphQLError: Type Query must define one or more fields
   @Query(() => Boolean)
   contactForm(): boolean {
```
Comment (Elweyn) [inline backend/src/graphql/resolvers/ContactFormResolver.ts:19]: Do we still need this query for the application?

**Review by Elweyn — COMMENTED** [review]
Seems good for me, lets just talk about the opened discussions.

File: backend/src/graphql/resolvers/ContactFormResolver.spec.ts [hunk 4]
```diff
@@ -195,25 +194,24 @@ weils nach Datum, Medium, Anlass und Kosten auflisten)?`,
         })
       })
 
-      it('has the contact form stored in the database', async () => {
-        const result: ContactForm[] = await prisma.contactForm.findMany()
+      it('calls sendContactFormEmail', () => {
+        expect(sendContactEmails).toHaveBeenCalled()
```
Comment (Mogge) [inline backend/src/graphql/resolvers/ContactFormResolver.spec.ts:198]:
```suggestion
        expect(sendContactEmails).toHaveBeenCalledWith({
          data: {
              firstName: 'Peter',
              lastName: 'Lustig',
              content: 'Hello DreamMall!',
              email: 'peter@lustig.de',
          },
        })
```

File: backend/test/helpers.ts [hunk 5]
```diff
@@ -4,6 +4,7 @@ export const deleteAll = async () => {
   await prisma.contactForm.deleteMany()
   await prisma.$executeRaw`DELETE FROM NewsletterPreOptIn`
   await prisma.$executeRaw`DELETE FROM NewsletterSubscription`
+  await prisma.event.deleteMany()
```
Comment (Mogge) [inline backend/test/helpers.ts:7]: No possible soft delete on events I suppose

**Review by Mogge — CHANGES_REQUESTED** [review]
Just some small improvement to a test

File: backend/test/helpers.ts (hunk 5 above)
Comment (ulfgebhardt) [reply backend/test/helpers.ts:7]: We will never delete events - there is no point in soft deletes here in my opinion

**Review by Mogge — APPROVED** [review]

**Pull request merged**